From 2f723de1cb76b90f7893bc24594590c78b22a077 Mon Sep 17 00:00:00 2001
From: yuric <yuric@marvell.com>
Date: Thu, 13 Dec 2018 14:44:27 +0200
Subject: [PATCH 04/30] net/mvpp2: fix stack corruption

Fixes stack corruption in mrvl_fill_bpool function
in case num > MRVL_PP2_RXD_MAX

Change-Id: I9398a0ccfa05f24905d4c7e8b00daab1bc48a993
Signed-off-by: yuric <yuric@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61749
Reviewed-by: Liron Himi <lironh@marvell.com>
Tested-by: Liron Himi <lironh@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1182
Tested-by: Liron Himi
Reviewed-by: Liron Himi
Reviewed-by: Alan Winkowski
---
 drivers/net/mvpp2/mrvl_ethdev.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/mvpp2/mrvl_ethdev.c b/drivers/net/mvpp2/mrvl_ethdev.c
index ab4c14e..f9eacf2 100644
--- a/drivers/net/mvpp2/mrvl_ethdev.c
+++ b/drivers/net/mvpp2/mrvl_ethdev.c
@@ -1555,8 +1555,8 @@ mrvl_vlan_filter_set(struct rte_eth_dev *dev, uint16_t vlan_id, int on)
 static int
 mrvl_fill_bpool(struct mrvl_rxq *rxq, int num)
 {
-	struct buff_release_entry entries[MRVL_PP2_RXD_MAX];
-	struct rte_mbuf *mbufs[MRVL_PP2_RXD_MAX];
+	struct buff_release_entry entries[num];
+	struct rte_mbuf *mbufs[num];
 	int i, ret;
 	unsigned int core_id;
 	struct pp2_hif *hif;
-- 
2.7.4

